db/crud.go

```diff
@@ -1421,6 +1421,8 @@ func (db *Database) updateAndReturnDoc(docid string, allowImport bool, expiry ui
 				return
 			}
 			docExists := currentValue != nil
+			doc.migrationRequired = true
```

Comment:
As discussed, can move this into the unmarshal, and only set when the xattr is nil.